test/e2e/policies.test-suite/balloons/n4c16/test22-isolcpus/code.var.sh

```diff
@@ -0,0 +1,33 @@
+helm-terminate
+helm_config=${TEST_DIR}/balloons-isolcpus.cfg helm-launch balloons
+
+vm-command "grep isolcpus=0,1 /proc/cmdline" || {
+    if [ "$distro" = "generic/fedora37" ]; then
+        fedora-set-kernel-cmdline "isolcpus=0,1"
+    else
+        ubuntu-set-kernel-cmdline "isolcpus=0,1"
+    fi
+    vm-reboot
+    vm-command "grep isolcpus=0,1 /proc/cmdline" || {
+        error "failed to set isolcpus kernel commandline parameter"
+    }
+    vm-command "systemctl restart kubelet"
+    sleep 5
+    vm-wait-process --timeout 120 kube-apiserver
+}
+vm-command "kubectl create namespace isolcpus"
```

Comment:
Two namespaces are created but not used. These should be cleaned up when exiting the test script.

I'd suggest changing the balloons configuration so that preferIsolCpus would be used in a balloon type that is not applied to all containers in the default namespace. (I'd suggest this because placing all default stuff to isolcpus sounds suspicious configuration, and I would not like to use a suspicious configuration even in an e2e test.)

If you change balloons-isolcpus.cfg so that only containers in the isolcpus namespace will be put into the isolcpus balloon, then there is use for this namespace and the configuration will be a better example, too.

Reply:
> Two namespaces are created but not used. These should be cleaned up when exiting the test script.
> 

Good catch. I will remove them.

> I'd suggest changing the balloons configuration so that preferIsolCpus would be used in a balloon type that is not applied to all containers in the default namespace. (I'd suggest this because placing all default stuff to isolcpus sounds suspicious configuration, and I would not like to use a suspicious configuration even in an e2e test.)
> 

Running pods in any namespaces than `kube-system`, `default` requires patching RBAC. For that reason I decided to deploy the workload in the `default` namespaces so that I don't have to introduce extra changes to RBAC.
If you think there is no difference between running the workload in `default` namespace vs `foo`, because we can just imagine that `default` namespace is a special namespace in this particular test that is meant to host only specific workload that requires isolated CPUs. Because, at the time of deploying workload on `default` namespace, there is no other Pod running on that namespace and as such I think it is safe to consider this namespace as special. But if there is any other reason to not pick the `default` namespace, I can extend the RBAC permissions.

Reply:
> > Two namespaces are created but not used. These should be cleaned up when exiting the test script.
> 
> Good catch. I will remove them.
> 
> > I'd suggest changing the balloons configuration so that preferIsolCpus would be used in a balloon type that is not applied to all containers in the default namespace. (I'd suggest this because placing all default stuff to isolcpus sounds suspicious configuration, and I would not like to use a suspicious configuration even in an e2e test.)
> 
> Running pods in any namespaces than `kube-system`, `default` requires patching RBAC. For that reason I decided to deploy the workload in the `default` namespaces so that I don't have to introduce extra changes to RBAC.

I don't think this is true for our e2e test setup. There the test user is running with the same kubeconfig as the cluster admin who bootstrapped the cluster. We have several test cases that create pods in the `kube-system` namespace successfully without any extra/special authorization.

Reply:
Have you tried in any other namespace which not kube-system or default  ?

Reply:
Unless the test shell library are broken and it does not really use namespaces in those cases...

Reply:
Which does not seem to be the case... sampling a test run I got among other things this. You can see a non-default, non-kube-system namespace pod there (in ns 'three'):
```
[vagrant@n4c16-generic-fedora37-containerd ~]$ kubectl get pods -A
NAMESPACE     NAME                                                        READY   STATUS        RESTARTS      AGE
default       pod1                                                        2/2     Running       0             7s
default       pod2                                                        1/1     Running       0             5s
kube-system   cilium-hhxcw                                                1/1     Running       1 (97s ago)   27d
kube-system   cilium-operator-6df6cdb59b-bgdzm                            1/1     Running       1 (97s ago)   27d
kube-system   coredns-7db6d8ff4d-gwf6x                                    1/1     Running       1 (97s ago)   27d
kube-system   coredns-7db6d8ff4d-r2c6l                                    1/1     Running       1 (97s ago)   27d
kube-system   etcd-n4c16-generic-fedora37-containerd                      1/1     Running       1 (97s ago)   27d
kube-system   kube-apiserver-n4c16-generic-fedora37-containerd            1/1     Running       1 (97s ago)   27d
kube-system   kube-controller-manager-n4c16-generic-fedora37-containerd   1/1     Running       1 (97s ago)   27d
kube-system   kube-proxy-b6ms9                                            1/1     Running       1 (97s ago)   27d
kube-system   kube-scheduler-n4c16-generic-fedora37-containerd            1/1     Running       1 (97s ago)   27d
kube-system   nri-resource-policy-balloons-wgcmp                          1/1     Running       0             12s
kube-system   pod0                                                        2/2     Terminating   0             9s
three         pod3                                                        2/2     Running       0             3s
```